DEVELOPMENT.md

```diff
@@ -38,4 +38,20 @@ for gapic-generator-java's Bazel build.
 
     ```sh
     mvn fmt:format
-    ```
\ No newline at end of file
+    ```
+
+## Testing guide
+There are 4 layers of testing in this repo:
+1. Traditional unit tests
+2. Golden unit tests
+3. Showcase integration tests
+4. Golden integration tests(We should stop adding new ones, and rely on golden unit tests and showcase tests)
```

Comment:
nit: I think this makes it sound like there is a possibility/ use case to add in the occasional golden integration tests.

Is there ever a use case/ need to add in new goldens? If not, I think we should just change it from `We should` to `Do not add new ones ...` and specify that we want to focus on adding new showcase tests while making the current goldens work.

Or if there is, maybe it would help to list some examples for when we should add goldens.

Reply:
> Is there ever a use case/ need to add in new goldens? If not, I think we should just change it from We should to Do not add new ones ... and specify that we want to focus on adding new showcase tests while making the current goldens work.

Great question and I don't have a definite answer, we need to audit the current golden integration tests and see if there is any cases that cannot be covered by unit or showcase tests. I cannot think a good exception for golden integration tests but I also don't want to say we don't need it at all without auditing.  So I prefer to keep it as `should` but maybe we can add more details to clarify.